syntax/gleam.vim

```diff
@@ -8,32 +8,65 @@ let b:current_syntax = "gleam"
 
 " Keywords
 syntax keyword gleamKeyword
-  \ module import pub panic use
-  \ type let as if else todo const
-  \ case assert tuple try opaque
+  \ import pub panic use type let as if else todo const case assert try opaque
 highlight link gleamKeyword Keyword
 
 " Function definition
 syntax keyword gleamDef fn nextgroup=gleamFunctionDef skipwhite skipempty
 highlight link gleamDef Keyword
 
-syntax match gleamFunctionDef "[a-z_-][0-9a-z_-]*" contained skipwhite skipnl
+syntax match gleamFunctionDef "[a-z][a-z0-9_]*\ze\s*(" skipwhite skipnl
+highlight link gleamFunctionDef Function
 highlight link gleamFunctionDef Function
 
-" Int
-syntax match gleamInt '\<\(0x[a-fA-F0-9_]\+\|[0-9][0-9_]*\)\>'
+" Number
+"" Int
+syntax match gleamInt '\<\d[0-9_]*\>'
 highlight link gleamInt Number
 
-" Float
-syntax match gleamFloat '\<[0-9][0-9_]*\.[0-9_]*\>'
+"" Binary
+syntax match gleamBinary '\<0[bB][01]+\>'
+highlight link gleamBinary Number
+
+"" Octet
+syntax match gleamOctet '\<0[oO][0-7]+\>'
+highlight link gleamOctet Number
+
+"" Hexadecimal
+syntax match gleamHexa '\<0[xX]\x+\>'
+highlight link gleamHexa Number
```

Comment:
These are not the int formats in Gleam

Reply:
Yes, thanks !
I just thought binaries, octets and hexadecimals could not have underscores, but that's not true.
They can, but only after a non-zero digit, and that's what I did not spot.
```gleam
let _binary_wrong = 0b0_1
let _binary_right = 0b1_1
```